test/test-rotateElement.js

```diff
@@ -0,0 +1,390 @@
+        let expectedResult = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            -----
+            -----
+            X----
+        `)
+        assert.deepEqual(expectedResult[0], stepElementTetromino)
+    })
+
+    it("element move the right field 4x7", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            -----
+            -----
+            X----
+        `)
+
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -X---
+            -----
+            -----
+            -----
+        `)
+
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            ----X
+            -----
+            -----
+        `)
+        assert.deepEqual(expectedResult[0], stepElementTetromino)
+    })
+
+    it("element move the top field 4x7", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -----
+            ----X
+            -----
+            -----
+        `)
+
+        let pivot = tetris.getCoordinates(`
+            -----
+            -----
+            -----
+            -X---
+            -----
+            -----
+            -----
+        `)
+
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            --X--
+            -----
+            -----
+            -----
+            -----
+            -----
+            -----
+        `)
+        assert.deepEqual(expectedResult[0], stepElementTetromino)
+    })
+})
+describe("Rotation where starts to the left,on the diagonal, has negative coordinates", function () {
+    it("element move the right on the diagonal field 3x5 ", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            X----
+        `)
+        let pivot = tetris.getCoordinates(`
+            --X--
+            -----
+            -----
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = tetris.getCoordinates(`
+            -----
+            -----
+            ----X
+        `)
+    })
+
+    it("element move the top on the diagonal field 3x5 ", function () {
+        let elementTetromino = tetris.getCoordinates(`
+            -----
+            -----
+            ----X
+        `)
+        let pivot = tetris.getCoordinates(`
+            --X--
+            -----
+            -----
+        `)
+        let stepElementTetromino = tetris.rotateElement(elementTetromino[0], pivot[0])
+        let expectedResult = [{ x: 4, y: -2 }]
```

Comment:
When you specify coordinates in non-graphical way - please use only object, no need to wrap it in the array

Reply:
Apply to all cases where you specify coordinates manually